lib/spack/spack/relocate.py

```diff
@@ -70,14 +70,13 @@ def get_patchelf():
     if patchelf is not None:
         return patchelf.path
     else:
-        if str(spack.architecture.platform()) == 'test':
-            return None
-        if str(spack.architecture.platform()) == 'darwin':
-            return None
         patchelf_spec = spack.cmd.parse_specs("patchelf", concretize=True)[0]
         patchelf = spack.repo.get(patchelf_spec)
         if not patchelf.installed:
-            patchelf.do_install(use_cache=False)
+            if platform.system().lower() == 'darwin':
+                patchelf.do_fake_install()
```

Comment:
I put this in because I was not checking the spec arch correctly in the test environment and relocate_elf_binary was being called which called get_patchelf. I would switch the check str(spack.architecture.platform()) == 'test' but I want to be able to install patchelf when the test runs on linux. I will think of another way to fix when the fake install is run.